ecs-agent/acs/session/session.go

```diff
@@ -262,8 +274,13 @@ func (s *session) startSessionOnce(ctx context.Context) error {
 		})
 		return err
 	}
+	acsConnectionMetric.Done(err)
```

Comment:
This ACS metric won't fire if `client.Connect` returns a non-nil error. Is that intentional? If so, this behavior is inconsistent with the DiscoverPollEndpoint metric.

Reply:
Because ACS infinitely retries its connection, is this metric going to be too noisy if we fire a failure metric on every failed connection?